docker/docker-compose.yaml

```diff
@@ -1,33 +1,41 @@
 # note: we do not run prod service with docker-compose, we use it just for
 # configuring the production build
 services:
-  prod:
-    # image name, both locally and public
-    image: opencodelists
+  # base service, exists to hold common config, but is not actually used directly
+  base:
     build:
-      context: ..
-      # path relative to context
-      dockerfile: docker/Dockerfile
-      # the prod stage in the Dockerfile
-      target: opencodelists-prod
-      # should speed up the build in CI, where we have a cold cache
-      cache_from:  # should speed up the build in CI, where we have a cold cache
-        - ghcr.io/opensafely-core/base-docker
-        - ghcr.io/opensafely-core/opencodelists
-      args:
-        # this makes the image work for later cache_from: usage
-        - BUILDKIT_INLINE_CACHE=1
-        # env vars supplied by just
-        - BUILD_DATE
-        - GITREF
+        context: ..
+        # path relative to context
+        dockerfile: docker/Dockerfile
+        # the prod stage in the Dockerfile
+        target: opencodelists-prod
+        # should speed up the build in CI, where we have a cold cache
+        cache_from:  # should speed up the build in CI, where we have a cold cache
+          - ghcr.io/opensafely-core/base-docker
+          - ghcr.io/opensafely-core/opencodelists
+        args:
+          # this makes the image work for later cache_from: usage
+          - BUILDKIT_INLINE_CACHE=1
+          # env vars supplied by just
+          - BUILD_DATE
+          - GITREF
     # use dockers builitin PID daemon
     init: true
     ports:
-      - "8000:8000"
+      - "8000:7000"
```

Comment:
I have mild preference to map ports through 1:1, not change them if possible, so would prefer `7000:7000`. I think it reduces confusion about what port something is running on if its the same inside and outside docker.

But I may well be missing something.

Reply:
I initially did that, but then changed it back so that it doesn't change under anyone who's working on opencodelists locally.  But that's probably only me really, so :woman_shrugging:

Reply:
Ah, right. To me, the ability to run both the docker version *and* the non-docker version simultaneously feels like a massive footgun. I almost quite like the simple port limit meaning they can not be run simultaneous. Especially if they are using the same db!

Reply:
I have updated it for docker and non-docker local servers